lib/puppet/provider/group/windows_adsi.rb

```diff
@@ -66,11 +66,9 @@ def group
   end
 
   def members
-    @members ||= Puppet::Util::Windows::ADSI::Group.name_sid_hash(group.members)
-
-    # @members.keys returns an array of SIDs. We need to convert those SIDs into
-    # names so that `puppet resource` prints the right output.
-    members_to_s(@members.keys).split(',')
+    group.members.map do |member|
+      member.to_s
+    end
   end
```

Comment:
This feels wrong to me. I would expect `members` to return the canonical SID so that the insync check is done against SIDs, but then modify the `is_to_s` method to emit account names (using `ConvertSidToSidString`)

Reply:
This will mean I will need to:
1. return member.sid array in `members` method
2. members_insync? will receive in current already the SID list does not have to  call name_sid_hash/etc.
3. is_to_s should handle members.sid array and convert it to names

Reply:
Perhaps a better way would be to change the `name_sid_hash(names)` method in `lib/puppet/util/windows/adsi/rb` as that is (probably) where the error is thrown.

And **add** a behaviour to ignore unresolvable, instead of throwing.

Reply:
Yes, probably I should not touch this method for the problem I want to solve.

But while digging to understand how this works I saw that `group.members` is an array of `Principal` on which we call `name_sid_hash` what will call `name_to_principal` on each member and it will convert `Principal` to string (`to_s`) and convert again to `Principal`.

Reply:
By try and error, I think there is also a 4: overwrite `retrieve` method for `members` to not dirrectly return members value, but `members_to_s(members)`.